src/Framework/Sdk/SdkReference.cs

```diff
@@ -102,8 +103,8 @@ public override int GetHashCode()
             unchecked
             {
                 var hashCode = (Name != null ? Name.GetHashCode() : 0);
```

Comment:
Should this not be lowercased too? You use a case-insensitive comparison for equality above.